quinn-proto/src/connection/mod.rs

```diff
@@ -3290,12 +3290,11 @@ impl Connection {
 
     fn set_peer_params(&mut self, params: TransportParameters) {
         self.streams.set_params(&params);
-        self.idle_timeout = match (self.config.max_idle_timeout, params.max_idle_timeout) {
-            (None, VarInt(0)) => None,
-            (None, x) => Some(x),
-            (Some(x), VarInt(0)) => Some(x),
-            (Some(x), y) => Some(cmp::min(x, y)),
-        };
+        self.idle_timeout = negotiate_max_idle_timeout(
+            self.config.max_idle_timeout.unwrap_or_default(),
```

Comment:
Nit: I think I'd prefer avoiding the `unwrap_or_default()` here in favor of explicitly matching `Some(VarInt(0)) | None`?

Reply:
Could you please clarify you mean change sinagure of `negotiate_max_idle_timeout` to accept both or only local argument as `Option<VarInt>` and do all checks inside or having something like outside:
```
match config.max_idle_timeout {
    None | Some(VarInt(0)) => None,
    Some(dur) => Some(dur)),
}
```
?

Reply:
Changed signature of `negotiate_max_idle_timeout` to accept both arguments as `Option<VarInt>` and perform checks inside to treat `Some(VarInt(0))` and `None` as disabled timeout.

Reply:
LGTM, thanks! Would you mind squashing the review feedback commit into the original one(s)?

Reply:
Squashed MR to single commit.

Reply:
Well, I actually liked the split you had previously between the commutative negotiation and changing the timeout type, but I can live with this. Should at least fix up the first line of the commit message to be more descriptive of the change though.

Reply:
Ah, sorry, the initial split was still imperfect because both commits mixed change of type and signature& implementation of negotiation function.
I've also update commit message to make it more informative so hope it's now clear what was the purpose of the commit for anyone who will see it in git history.